contracts/mock/MockOracle.sol

```diff
@@ -40,5 +41,10 @@ contract MockOracle is IOracle {
 
     function setExchangeRate(uint256 usdPerEth) public {
         _usdPerEth = usdPerEth;
+        price = Decimal.from(usdPerEth);
+    }
+
+    function setExchangeRateScaledBase(uint256 usdPerEth) public {
+        price = Decimal.D256({ value: usdPerEth });
```

Comment:
I think it would be easier to have this function take in two unscaled parameters and use a ratio maybe? Nbd